### DISCUSSION TITLE
How to clearMediaCollection without deleting files?

### DISCUSSION THREAD

**Discussion by johncarter-** [spatie/laravel-medialibrary#2743] [Q&A]
Hi

I have a situation with user profile photos, where I want the user to be able to remove the profile from their account, but I don't want to delete it because there will be historic references to that path (e.g. old database notifications).

I thought I could do something like this: `$this->user->media('profiles')->each(fn (Media $media) => $media->model()->dissociate()->save());` 

For this to work, however, I would need to publish and modify the default migration to make the `model_type` and `model_id` nullable which seems acceptable.

I just wanted to know if any one has a better way of removing media from a collection without actually deleting the file?

**Comment by erikn69** [comment]
Try `Media::whereCollectionName('profiles')->whereModelTypeAndModelId('your_model','your_id')->delete();`

**Comment by squiaios** [comment]
A better solution is to call the `deletePreservingMedia()` method on your user;

https://spatie.be/docs/laravel-medialibrary/v9/basic-usage/retrieving-media
